src/commandsAndMenu.tsx

```diff
@@ -410,6 +410,58 @@ export function addCommands(
     }
   });
 
+  /** Add git reset --hard <remote-tracking-branch> command */
+  commands.addCommand(CommandIDs.gitResetToRemote, {
+    label: trans.__('Reset to Remote'),
+    caption: trans.__('Reset Current Branch to Remote State'),
+    isEnabled: () => gitModel.pathRepository !== null,
+    execute: async _ => {
+      const result = await showDialog({
+        title: trans.__('Reset to Remote'),
+        body: (
+          <span>
+            {trans.__(
+              'To bring the current branch to the state of its corresponding remote tracking branch, \
+              a hard reset will be performed, which may result in some files being permanently deleted \
+              and some changes being permanently discarded. Are you sure you want to proceed? \
+              This action cannot be undone.'
+            )}
+          </span>
```

Comment:
Thinking again and related to #1086, it will be nice to add a checkbox _Close all opened files to avoid conflicts_ (that will be checked by default) in the dialog. Then if it is checked, close all opened files before the reset action.

Tip for the implementation, add a callback on the check box that change a local variable to the `execute` function.